wx/tools/wxget.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#----------------------------------------------------------------------
+# Name:        wx.tools.wxget
+# Purpose:     wx Based alternative to wget
+#
+# Author:      Steve Barnes
+#
+# Created:     06-Aug-2017
+# Copyright:   (c) 2017 by Steve Barnes
+# Licence:     wxWindows license
+# Tags:        phoenix-port, py3-port
+#
+# Module to allow cross platform downloads originally from answers to:
+# https://stackoverflow.com/questions/22676/how-do-i-download-a-file-over-http-using-python
+# by Stan and PabloG then converted to wx.
+#----------------------------------------------------------------------
+
+"""
+wxget.py -- wx Version of wget utility for platform that don't have it already.
+
+Usage:
+    wxget URL [DEST_DIR]
+
+Where URL is a file URL and the optional DEST_DIR is a destination directory to
+download to, (default is to prompt the user).
+"""
+from __future__ import (division, absolute_import, print_function, unicode_literals)
+
+import sys
+import os
+import wx
+
+if sys.version_info >= (3,):
+    from urllib.error import HTTPError
+    import urllib.request as urllib2
+    import urllib.parse as urlparse
+else:
+    import urllib2
+    from urllib2 import HTTPError
+    import urlparse
+
+def get_docs_demo_url(demo=False):
+    """ Get the URL for the docs or demo."""
+    if demo:
+        pkg = 'demo'
+    else:
+        pkg = 'docs'
+    base_url = "https://extras.wxpython.org/wxPython4/extras/%s/wxPython-%s-%s.tar.gz"
+    ver = wx.version().split(' ')[0]
+    major = ver.split('.')[0]
+    if major != '4':
+        raise ValueError("wx Versions before 4 not supported!")
+    return base_url % (ver, pkg, ver)
+
+def get_save_path(url, dest_dir, force=False):
+    """ Get the file save location."""
+    old_dir = os.getcwd()
+    if not dest_dir:
+        dest_dir = os.getcwd()
+    else:
+        if not os.path.exists(dest_dir):
+            os.makedirs(dest_dir)
+        os.chdir(dest_dir)
+    filename = os.path.basename(urlparse.urlsplit(url)[2])
+    if not filename:
+        filename = 'downloaded.file'
+
+    if not force:
+        dlg = wx.FileDialog(
+            None, message="Save As ...", defaultDir=dest_dir,
+            defaultFile=filename, style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT
+        )
+        if dlg.ShowModal() == wx.ID_OK:
+            dest_dir, filename = os.path.split(dlg.GetPath())
+        else:
+            url = None
+        del dlg
```

Comment:
`del dlg` is not enough, you need to actually call `dlg.Destroy()`. Or you can use a context manager, which will call `Destroy` for you on the way out, like this:

```python
    with wx.FileDialog(...) as dlg:
        if dlg.ShowModal() == ...
```